vertx-config/src/main/java/io/vertx/config/ConfigRetriever.java

```diff
@@ -0,0 +1,97 @@
+package io.vertx.config;
+
+import io.vertx.codegen.annotations.CacheReturn;
+import io.vertx.codegen.annotations.VertxGen;
+import io.vertx.config.spi.ConfigStore;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.config.impl.ConfigRetrieverImpl;
+
+/**
+ * Defines a configuration retriever that read configuration from
+ * {@link ConfigStore}
+ * and tracks changes periodically.
+ *
+ * @author <a href="http://escoffier.me">Clement Escoffier</a>
+ */
+@VertxGen
+public interface ConfigRetriever {
+
+  /**
+   * Creates an instance of the default implementation of the {@link ConfigRetriever}.
+   *
+   * @param vertx   the vert.x instance
+   * @param options the options, must not be {@code null}, must contain the list of configured store.
+   * @return the created instance.
+   */
+  static ConfigRetriever create(Vertx vertx, ConfigRetrieverOptions options) {
+    return new ConfigRetrieverImpl(vertx, options);
+  }
+
+  /**
+   * Creates an instance of the default implementation of the {@link ConfigRetriever}, using the default
+   * settings (json file, system properties and environment variables).
+   *
+   * @param vertx   the vert.x instance
+   * @return the created instance.
+   */
+  static ConfigRetriever create(Vertx vertx) {
+    ConfigRetrieverOptions options = new ConfigRetrieverOptions();
+    options
+        .addStore(
+            new ConfigStoreOptions().setType("json")
+                .setConfig(vertx.getOrCreateContext().config()))
+        .addStore(
+            new ConfigStoreOptions().setType("sys")
+        )
+        .addStore(new ConfigStoreOptions().setType("env")
+        );
+    return create(vertx, options);
+  }
+
+  /**
+   * Reads the configuration from the different {@link ConfigStore}
+   * and computes the final configuration.
+   *
+   * @param completionHandler handler receiving the computed configuration, or a failure if the
+   *                          configuration cannot be retrieved
+   */
+  void getConfig(Handler<AsyncResult<JsonObject>> completionHandler);
+
+  /**
+   * Same as {@link #getConfig(Handler)}, but returning a {@link Future} object. The result is a
+   * {@link JsonObject}.
+   */
+  Future<JsonObject> getConfigFuture();
```

Comment:
In the initial version I did. The feedback I have received contain usage of this method (from different users).

Removing this method would mean that they can't move to the Vert.x version without rewriting their code differently. 

To mitigate the issue I can move this method as a static method to a helper class (you still need to pass the instance of the retriever as param, so it would break but the adaptation is straightforward).